types/tls.go

```diff
@@ -106,67 +78,78 @@ func (v *TLSVersion) UnmarshalJSON(bytes []byte) error {
-	TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256                        = 0xc02f
-	TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256                  = 0xcca8 // Not used by Consul agent yet
-	TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA                         = 0xc009
-	TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA                           = 0xc013
-	TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384                      = 0xc02c
-	TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384                        = 0xc030
-	TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA                         = 0xc00a
-	TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA                           = 0xc014
-
-	// Older cipher suites not supported for Consul agent TLS, will eventually be removed from Envoy defaults
-	TLS_RSA_WITH_AES_128_GCM_SHA256 = 0x009c
-	TLS_RSA_WITH_AES_128_CBC_SHA    = 0x002f
-	TLS_RSA_WITH_AES_256_GCM_SHA384 = 0x009d
-	TLS_RSA_WITH_AES_256_CBC_SHA    = 0x0035
+	// Cipher suites used by Envoy but not used by Consul agent yet
+	TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256 = "TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256"
+	TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256   = "TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256"
+
+	// Cipher suites used by both Envoy and Consul agent
+	TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 = "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256"
+	TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256   = "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256"
+	TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA    = "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA"
+	TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA      = "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA"
+	TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384 = "TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384"
+	TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384   = "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384"
+	TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA    = "TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA"
+	TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA      = "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA"
+
+	// Older cipher suites not supported for Consul agent TLS,
+	// will eventually be removed from Envoy defaults
+	TLS_RSA_WITH_AES_128_GCM_SHA256 = "TLS_RSA_WITH_AES_128_GCM_SHA256"
+	TLS_RSA_WITH_AES_128_CBC_SHA    = "TLS_RSA_WITH_AES_128_CBC_SHA"
+	TLS_RSA_WITH_AES_256_GCM_SHA384 = "TLS_RSA_WITH_AES_256_GCM_SHA384"
+	TLS_RSA_WITH_AES_256_CBC_SHA    = "TLS_RSA_WITH_AES_256_CBC_SHA"
 
 	// Additional cipher suites used by Consul agent but not Envoy
 	// TODO: these are both explicitly listed as insecure and disabled in the Go source, should they be removed?
 	// https://cs.opensource.google/go/go/+/refs/tags/go1.17.3:src/crypto/tls/cipher_suites.go;l=329-330
-	TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256 = 0x0023
-	TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256   = 0xc027
+	TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256 = "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256"
+	TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256   = "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256"
 )
 
 var (
-	TLSCipherSuites = map[string]TLSCipherSuite{
+	ConsulAgentTLSCipherSuites = map[string]TLSCipherSuite{
```

Comment:
Are all of these keys and values basically the same strings on both sides? This map could also be switched to `map[TLSCipherSuite]struct{}` or `map[string]struct{}` (with the latter requiring a simple string(v) cast to lookup TLSCipherSuites directly)